lib/Doctrine/ODM/MongoDB/UnitOfWork.php

```diff
@@ -1297,7 +1298,6 @@ private function cascadePostUpdateAndPostPersist(ClassMetadata $class, $document
                     } else {
                         if ( ! empty($entryClass->lifecycleCallbacks[Events::postUpdate])) {
                             $entryClass->invokeLifecycleCallbacks(Events::postUpdate, $entry);
-                            $this->recomputeSingleDocumentChangeSet($entryClass, $entry);
```

Comment:
To me it always seemed like user can call `recomputeSingleDocumentChangeSet` from his listener since he gets DocumentManager in the event argument (and knows if he needs recomputing) but lifecycle callbacks are not getting any information so we assume they might have changed something in document and are recomputing to see if that's true?